### PR TITLE
Component new navigation

### PR DESCRIPTION
**Pull request by boxadesign**
### What is the context of this PR?
Added header navigation to the pattern library.

### How to review 
Check code, accessibility and mobile functionality.

### CODE REVIEW

**Comment by boxadesign** [comment]
@bennpearson Thought I'd dealt with that but you're right.

**Comment by armstrongb** [comment]
It appears impossible to tab to the main navigation (Surveys | Messages) on desktop.
(https://deploy-preview-111--sdc-global-design-patterns.netlify.com/components/preview/header-navigation--main.html)

**Comment by armstrongb** [comment]
We may run into A11y problems with the `:hover` state on the main navigation link (https://deploy-preview-111--sdc-global-design-patterns.netlify.com/components/preview/header-navigation--main.html) as it is such a subtle change.

**Comment by armstrongb** [comment]
Tabbing on desktop is now working again 👍.

File: components/header/header.hbs [hunk 1]
```diff
@@ -1,8 +1,63 @@
...
+                    {{> @button btn-classes='btn--ghost print__hidden' label=button}}
+                </div>
+                {{/if}}
+            </div>
+        </div>
+    </div>
+    {{#if nav}}  
+    <div class="header__nav">
+        <div class="container">
+            <nav class="nav nav--horizontal nav--light nav--main nav--h-m js-main-nav" aria-label="Main menu" id="main-nav">
```
Comment (armstrongb) [inline components/header/header.hbs]: In **Voiceover** this menu is announced irrespective of whether it is in use on these pages.

https://deploy-preview-111--sdc-global-design-patterns.netlify.com/components/preview/header-navigation--secondary.html

and

https://deploy-preview-111--sdc-global-design-patterns.netlify.com/components/preview/header-navigation--save.html

The main menu is read out in Voiceover after the `H1`.

`<nav ... aria-label="Main menu"` should not be included if no content is available.

**Comment by boxadesign** [comment]
@armstrongb Think all should be good on this one now.

**Comment by boxadesign** [comment]
@bennpearson @armstrongb I have removed the animation from the mobile nav menu. I don't think it added anything significant to the experience and taking it out removes all of the tabbing issues and extra js code for handling the tab index etc.

**Comment by boxadesign** [comment]
I'm closing this PR as the work has been incorporated (and changed quite a bit) in this PR -> https://github.com/ONSdigital/sdc-global-design-patterns/pull/129

**Pull request closed without merging**